lib/exchange.js

```diff
@@ -44,32 +51,68 @@ function exchange(name, type, options) {
   var ready = false;
   var connection, channel;
   var publishing = 0;
-  var replyQueue = queue({ exclusive: true });
+  var options = extend({}, DEFAULT_EXCHANGE_OPTIONS, options)
+  var replyQueue = !options.noReply && queue({ exclusive: true });
   var pendingReplies = {};
 
-  var emitter = _.extend(new EventEmitter(), {
+  var emitter = extend(new EventEmitter(), {
     name: name,
     type: type,
-    options: _.extend({}, DEFAULT_EXCHANGE_OPTIONS, options),
+    options: extend({}, DEFAULT_EXCHANGE_OPTIONS, options),
     queue: createQueue,
     connect: connect,
-    publish: publish
+    publish: publish,
+    rpcClient: rpcClient,
+    rpcServer: rpcServer
   });
 
   return emitter;
 
+  function rpcClient(key, msg, options, cb) {
+
+    if (!key) {
+      throw new Error('missing rpc method');
+    }
+
+    if (!cb && typeof options === 'function') {
+      cb = options;
+    }
+
+    if (!options || typeof options !== 'object') {
+      options = DEFAULT_RPC_CLIENT_OPTIONS;
+    }
+
+    var opts = extend({}, {
+      key: key,
+      rpcCallback: cb
+    }, options);
+
+    publish(msg, opts);
+  }
+
+  function rpcServer(key, handler) {
+    var rpcQueue = createQueue({
+      key: key,
+      name: key,
+      prefetch: 1,
+      durable: false,
+      autoDelete: true
+    });
+    rpcQueue.consume(handler);
+  }
+
   function connect(con) {
     connection = con;
     connection.createChannel(onChannel);
-    replyQueue.on('close', bail.bind(this));
-    replyQueue.consume(onReply, { noAck: true });
+    replyQueue && replyQueue.on('close', bail.bind(this));
```

Comment:
Would you mind changing this to simple conditional? The use of logical operators on an expression line could be confusing and as its written here it unnecessarily evaluates the condition twice.

Preferred: 

```
if (replyQueue) {
  replyQueue.on('close', bail.bind(this));
  replyQueue.consume(onReply, { noAck: true });
}
```